### DISCUSSION TITLE
Please HELP

### DISCUSSION THREAD

**Discussion by iospoisson** [laravel/framework#45840] [Q&A]
Hi everyone ,I am trying to change the layout of my booking page to get 2 responsive buttons one for phone and another one for whatsup
this is the original working code
  ```
</div>
                        <button type="button" class="btn btn-primary" @click="doSubmit($event)" :class="{'disabled':onSubmit}" name="submit">
                            <span >{{__("Book Now")}}</span>
                            <i v-show="onSubmit" class="fa fa-spinner fa-spin"></i>
                        </button>
                    </div>
```
if i change to 

```
</div>
                        <a class="btn btn-sm btn-primary" href="tel:{{ $vendor->phone}}" ><i class="icon ion_ios_telephone_outline"></i> {{__('Apelati acum')}}</a>
			            <a class="btn btn-sm btn-primary" href="https://api.whatsapp.com/send?phone={{ $vendor->phone}}&text=Buna ziua,va contactez de pe https://infoturist.acasalanoi.ro" ><i class="icon ion_ios_telephone_outline"></i> {{__('WhatsApp')}}</a>
                    </div>
```

i get 
500 Server Error

the log shows this

`Undefined variable $vendor {"view":{"view":"/home/acasalanoi/infoturist.acasalanoi.ro/themes/Mytravel/Hotel/Views/frontend/layouts/details/hotel-rooms.blade.php","data":{"errors":"<pre class=sf-dump id=sf-dump-983699632 data-indent-pad=\" \"><span class=sf-dump-note>Illuminate\\Support\\ViewErrorBag</span> {<a class=sf-dump-ref>#1496</a><samp data-depth=1 class=sf-dump-expanded>`

what i am doing wrong?
thanks

**Comment by henzeb** [comment]
well, it says it is trying to use a variable which is not defined. `$vendor->phone`

what is your code where you return the view?

**Comment by Kristories** [comment]
You should make sure that the `$vendor` is correctly defined and passed to the view : 

```php
...

$data['vendor'] = Vendor::first(); // example

return view('hotel-rooms', $data);

...
```
